src/set_theory/fixed_points.lean

```diff
@@ -0,0 +1,508 @@
+  (∀ i hi, f i hi b ≤ nfp_bfamily o f a) ↔ b ≤ nfp_bfamily o f a :=
+begin
+  refine ⟨λ h, _, λ h i hi, _⟩,
+  { have ho' : 0 < o := ordinal.pos_iff_ne_zero.2 ho,
+    exact ((H 0 ho').self_le b).trans (h 0 ho') },
+  rw ←nfp_bfamily_fp H i,
+  exact (H i hi).strict_mono.monotone h
+end
+
+theorem nfp_bfamily_eq_self {a} (h : ∀ i hi, f i hi a = a) : nfp_bfamily o f a = a :=
+nfp_family_eq_self (λ _, h _ _)
+
+/-- A generalization of the fixed point lemma for normal functions: any family of normal functions
+    has an unbounded set of common fixed points. -/
+theorem fp_bfamily_unbounded (H : ∀ i hi, is_normal (f i hi)) :
+  (⋂ i hi, function.fixed_points (f i hi)).unbounded (<) :=
+λ a, ⟨_, by { rw set.mem_Inter₂, exact λ _ _, nfp_bfamily_fp H _ _ _ },
+  (self_le_nfp_bfamily f a).not_lt⟩
+
+/-- The derivative of a family of normal functions is the sequence of their common fixed points. -/
+def deriv_bfamily (o : ordinal) (f : Π b < o, ordinal → ordinal) : ordinal → ordinal :=
+deriv_family (family_of_bfamily o f)
+
+theorem deriv_bfamily_eq_deriv_family {o : ordinal} (f : Π b < o, ordinal → ordinal) :
+  deriv_bfamily o f = deriv_family (family_of_bfamily o f) :=
+rfl
+
+theorem deriv_bfamily_is_normal {o : ordinal} (f : Π b < o, ordinal → ordinal) :
+  is_normal (deriv_bfamily o f) :=
+deriv_family_is_normal _
+
+theorem deriv_bfamily_fp (H : ∀ i hi, is_normal (f i hi)) (a : ordinal) (i hi) :
+  f i hi (deriv_bfamily o f a) = deriv_bfamily o f a :=
+by { rw ←family_of_bfamily_enum o f, exact deriv_family_fp (λ _, H _ _) _ _ }
+
+theorem le_iff_deriv_bfamily (H : ∀ i hi, is_normal (f i hi)) {a} :
+  (∀ i hi, f i hi a ≤ a) ↔ ∃ b, deriv_bfamily o f b = a :=
+begin
+  unfold deriv_bfamily,
+  rw ←le_iff_deriv_family,
+  { refine ⟨λ h i, h _ _, λ h i hi, _⟩,
+    rw ←family_of_bfamily_enum o f,
+    apply h },
+  exact λ _, H _ _
+end
+
+theorem fp_iff_deriv_bfamily (H : ∀ i hi, is_normal (f i hi)) {a} :
+  (∀ i hi, f i hi a = a) ↔ ∃ b, deriv_bfamily o f b = a :=
+begin
+  rw ←le_iff_deriv_bfamily H,
+  refine ⟨λ h i hi, le_of_eq (h i hi), λ h i hi, _⟩,
+  rw ←(H i hi).le_iff_eq,
+  exact h i hi
+end
+
+theorem deriv_bfamily_eq_enum_ord (H : ∀ i hi, is_normal (f i hi)) :
+  deriv_bfamily o f = enum_ord (⋂ i hi, function.fixed_points (f i hi)) :=
+begin
+  rw ←eq_enum_ord _ (fp_bfamily_unbounded H),
+  use (deriv_bfamily_is_normal f).strict_mono,
+  rw set.range_eq_iff,
+  refine ⟨λ a, set.mem_Inter₂.2 (deriv_bfamily_fp H a), λ a ha, _⟩,
+  rw set.mem_Inter₂ at ha,
+  rwa ←fp_iff_deriv_bfamily H
+end
+
+end
+
+/-! ### Fixed points of a single function -/
+
+section
+variable {f : ordinal.{u} → ordinal.{u}}
+
+/-- The next fixed point function, the least fixed point of the normal function `f`, at least `a`.
+-/
+def nfp (f : ordinal → ordinal) : ordinal → ordinal :=
+nfp_family (λ _ : unit, f)
+
+theorem nfp_eq_nfp_family (f : ordinal → ordinal) : nfp f = nfp_family (λ _ : unit, f) :=
```

Comment:
Should this be a simp-lemma in the opposite direction?

Reply:
This probably won't stay a simp lemma for long, since my subsequent PR generalizes this to nonempty types. But for now, sure.